Monika After Story/game/script-farewells.rpy

```diff
@@ -546,7 +546,41 @@ label bye_prompt_sleep:
 
     if 20 <= curr_hour < 24:
         # decent time to sleep
-        if mas_isMoniNormal(higher=True):
+        if mas_isMoniEnamored(higher=True) and persistent._mas_first_kiss is not None and random.randint(1, 3) == 1:
+            m 1rublsdla "Think I could...get a goodnight kiss?"
+            $ _history_list.pop()
+            menu:
+                m "Think I could...get a goodnight kiss?"
+                "Sure!":
+                    call monika_kissing_motion
+                    m 6ekbfa "Now I hope you'll dream better~"
+                    m 1hkbfb "Sleep tight!"
+
+                "Perhaps another time.":
+                    if random.randint(1, 3) == 1:
+                        m 2rublp "Aww, come on..."
+                        m 2nublu "I know you really want to~"
+                        show monika 5hublb at t11 zorder MAS_MONIKA_Z with dissolve_monika
+                        m 5hublb "Can I please get a goodnight kiss?"
+                        $ _history_list.pop()
+                        menu:
+                            m "Can I please get a goodnight kiss?"
+                            "Okay.":
+                                show monika 6dubfd at t11 zorder MAS_MONIKA_Z with dissolve_monika
+                                call monika_kissing_motion
+                                m 6ekbfa "Now I hope you'll dream better~"
+                                m 1hkbfb "Sleep tight!"
+                            "No.":
+                                $ mas_loseAffection()
+                                m 2dsd "..."
+                                m "Fine."
+                                m 2dsc "Goodnight [player]..."
+
+                    else:
+                        m 1wkblc "Aww... Okay, but you owe me one."
+                        m 1kkb "I love you! Sleep tight!"
```

Comment:
This should be turned into a label which can be called from places we want to use this.

You should aim to do the least amount of copy/paste as possible, instead opting to call a function/method/label to do the task instead. This way in case there's an issue in the code, we don't have to fix it 4 times but just once instead.

Also means using it in more places is easier, as it turns into just a single line.